memgpt/local_llm/utils.py

```diff
@@ -6,6 +6,24 @@
 import memgpt.local_llm.llm_chat_completion_wrappers.zephyr as zephyr
 
 
+# deprecated for Box
+class DotDict(dict):
```

Comment:
@sarahwooders I think if we're trying to be extra safe we should be checking if any objects loaded on a pickle load even are DotDict, and if so, cast it to Box

However, I don't think this is super urgent so I'm fine just doing this for now (also not sure if that is even needed).

Reply:
Wait why would anything be a `DotDict`?

Reply:
Prior to https://github.com/cpacker/MemGPT/pull/534, we were using `DotDict` (our own custom class) with local LLMs to mimic the object access patterns on ChatCompletion responses.

When we merged https://github.com/cpacker/MemGPT/pull/534, `DotDict` got stripped from the codebase. However I forgot that old pickle files of agents run with local LLMs would still have `DotDict` inside the message objects, hence this PR to add `DotDict` back to patch the runtime error.